server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java

```diff
@@ -61,7 +77,7 @@
   private final CompactionSegmentSearchPolicy policy;
   private final IndexingServiceClient indexingServiceClient;
 
-  private Object2LongOpenHashMap<String> totalSizesOfSegmentsAwaitingCompactionPerDataSource;
+  private AtomicReference<Map<String, AutoCompactionSnapshot>> autoCompactionSnapshotPerDataSource = new AtomicReference<>();
```

Comment:
Could you please add a Javadoc about the concurrent access pattern on `autoCompactionSnapshotPerDataSource`? I guess it can say "This variable is updated by the Coordinator thread executing duties and read by HTTP threads processing Coordinator API calls."

Reply:
nit: this variable can be final.

Reply:
Done